src/pages/docs/[[...filePath]].tsx

```diff
@@ -0,0 +1,198 @@
+    return (
+      <div style={{ margin: "32px auto", maxWidth: "1000px" }}>
+        {header}
+        <Typography style={{ margin: "32px auto", textTransform: "capitalize" }} variant="h2"> {filePath.slice(0, -1)} </Typography> {/* Print the subfolders we're in */}
+        <List>
+          {
+            pages.map((pageName) => (
+              <NextLink key={pageName} href={`/docs/${filePath}${pageName}`}>
+                <ListItem button key={pageName}>
+                  <Typography style={{ textTransform: "capitalize" }}>
+                    {
+                      pageName.replace(
+                        /-/g,
+                        " "
+                      ) /* we don't use the front matter of the file at this point to simplify loading, so we have to cleanup the name manually */
+                    }
+                  </Typography>
+                </ListItem>
+              </NextLink>
+            ))
+          }
+          <hr></hr>
+          {previousPageLink(filePath)}
+          {filePath === '' /* Back home if we're in /docs, back to /docs if we're in a subfolder */
+            ? homeLink
+            : indexLink
+          }
+        </List>
+      </div>
+    )
+  }
+}
+
+interface PathsProps {
+  params: { filePath: Array<String> } // Use String and not string to match vulcan-npm MdxPath syntax
+}
+export async function getStaticPaths() {
+  const docsDir = path.resolve("./src/content/docs"); // relative to the project root
+  const files = await getMdxPaths(docsDir);
+  // paths is the file without the extension, shaped as [{ params: { filePath: [ 'subfolder', 'file' ] } } ]
+  const paths = fillPaths(files);
+  return {
+    paths,
+    fallback: false,
+  };
+}
+function fillPaths(files: MdxPath[]): PathsProps[] {
+  const paths: Array<PathsProps> = [{ params: { filePath: [''] } }];
+  // add all subfolder paths
+  // example for the fileName "["subfolder1", "subfolder2", "file"]", push to paths ["subfolder1"], ["subfolder1", "subfolder2"] and ["subfolder1", "subfolder2", "file"]
+  files.forEach(file => {
+    file.params.fileName.forEach((item, index, array) => {
+      paths.push({ params: { filePath: array.slice(0, index + 1) } });
+    });
+  });
+  return paths;
+}
+
+/**
+ * Get the mdx files if the paths leads to a folder, or the content of the mdx file if it is.
+ * Be sure that a folder returns two elements and a file returns three, or change the implementation of the page consequently.
+ */
+export async function getStaticProps({ params }) {
+  // Check if the filePath is a file or a directory
+  if (!params.filePath) { // We're in /docs
+    return (await getMdxPages(path.resolve("./src/content/docs"), '')); // relative to the project root
+  }
+  else { // we're in a file or a subfolder
+    let resolvedPath = path.resolve("./src/content/docs/" + params.filePath.join('/'));
+    if (existsSync(resolvedPath) && lstatSync(resolvedPath).isDirectory()) { // We're in a subfolder
```

Comment:
Ok, but please be careful on sync stuff, they bloat your Node server. Node is better with async, getStaticProps can be an async function with no problem. Node fs provides an implementation with promises : https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api that will work with async await

Reply:
Understood. I tried this instead:
`try { // We're in a subfolder
      await fsPromises.stat(resolvedPath);
      return (await getMdxPages(resolvedPath, params.filePath.join('/')));
    } catch { // We're in a file`
I used a try/catch here because readFile() will be ok for folders but fail for files here since we use the path without the extension, but I'm not sure about using a catch block 50% of the time. Is there a way to do this clearer ?

Reply:
Ah yeah thats a classic: https://stackoverflow.com/questions/31799274/node-js-fs-exists-will-be-deprecated-what-to-use-instead
So yeah existsSync is ok, alternatives are quite verbose indeed I don't like the `.stat`. I think `exist` is rather fast, so it's ok to use the sync version, it won't really block.

Reply:
So the initial version sounds ok to me, you can leave it as is